documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
+
+**Description** :  Do not use case sensitivity to differentiate model names.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: In Python SDK, model names are converted to forms starting with capital. So all of "AAAA", "aaaa", "Aaaa" will be transformed to "Aaaa". So differentiating model names by their case sensitivities would break Python SDK generation.
+
+**How to fix the violation**: Rename the duplicate name .
+
+The following would be invalid:
+
+```json
+"definitions": {
+  "SKU": {
+    "type": "string",
+    "description": "SKU in request"
+  },
+  "sku": {
+    "type": "string",
+    "description": "SKU in response"
+  }
+}
+```
+
+The following would be valid:
+
+```json
+"definitions": {
+  "requestSKU": {
+    "type": "string",
+    "description": "SKU in request"
+  },
+  "responseSKU": {
+    "type": "string",
+    "description": "SKU in response"
+  }
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4034"></a>R4034 AzureResourceTagsSchemaValidation
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The resource "{0}" property tags schema is different with the common type definition.
+
+**Description** :  This rule is to check if the tags definition of a resource is same with the common tags definition.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: It will block the SDK generation for terraform, as it's only accepted that the Golang type for tags is map[string]*string .
+
+**How to fix the violation**: Please reference to the common tags definition in [v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/0e18f46fd2c210f85b5ec0f9dd9be664242bee82/specification/common-types/resource-management/v2/types.json#L146).
+
+The following would be invalid:
+
+```json
+"tags": {
+  "type": "object",
+  "description": "Resource Tags"
+}
+```
+
+The following would be valid:
+
+```json
+"tags": {
+  "type": "object",
+  "additionalProperties": {
+    "type": "string"
+  },
+  "x-ms-mutability": [
+    "read",
+    "create",
+    "update"
+  ],
+  "description": "Resource Tags"
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4035"></a>R4035 PrivateEndpointResourceSchemaValidation
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The private endpoint model "{0}" schema is different with the common type definition.
```

Comment:
The private endpoint model "{0}" does not follow/adopt the common type definition.

Reply:
or 'conform to' as being used below

Reply:
fixed